cclib/parser/gamessdatparser.py

```diff
@@ -0,0 +1,209 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2023, the cclib development team
+#
+# This file is part of cclib (http://cclib.github.io) and is distributed under
+# the terms of the BSD 3-Clause License.
+
+"""Parser for GAMESS(US) .dat output files"""
+
+
+import re
+import numpy
+
+from cclib.parser import logfileparser
+from cclib.parser import utils
+
+
+class GAMESSDAT(logfileparser.Logfile):
+    """A GAMESS .dat log file"""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(logname="GAMESS .dat", *args, **kwargs)
+
+    def __str__(self):
+        """Return a string representation of the object."""
+        return f"GAMESS .dat log file {self.filename}"
+
+    def __repr__(self):
+        """Return a representation of the object."""
+        return f'GAMESS .dat ("{self.filename}")'
+    
+    def normalisesym(self, label):
+        """Normalise the symmetries used by GAMESS .dat."""
+
+        pass
+
+    def before_parsing(self):
+        pass
+
+    def extract(self, inputfile, line):
+        """Extract information from the file object inputfile."""
+        
+        # Extract element and its properties
+
+        # $DATA  
+        # water                                                                           
+        # C1       0
+        # O           8.0      0.0000000000      0.0000000000      0.0000000000
+        #    STO     3
+                
+        # H           1.0      0.9900000000      0.0000000000      0.0000000000
+        #    STO     3
+                
+        # H           1.0     -0.2728810000      0.9516490000      0.0000000000
+        #    STO     3
+                
+        #  $END   
+
+        # Extract molecule name
+
+        if line[1:6] == "$DATA":
+
+            line = next(inputfile)
+            name = line.split('\n')[0].strip()
+            self.metadata["name"] = name
+
+        # Extract atomic information
+
+        atomic_data = []
+        line = next(inputfile)
+
+        while line.strip() != "$END":
+            parts = line.split()
+            if len(parts) >= 2:
+                symbol = parts[0]
+                mass = float(parts[1])
+                coordinates = [float(coord) for coord in parts[2:]]
+                atom_info = {"symbol": symbol, "mass": mass, "coordinates": coordinates}
+                atomic_data.append(atom_info)
+            line = next(inputfile)
+        
+        self.metadata["atoms"] = atomic_data
```

Comment:
populate `atomcoords`, `atommasses`, `atomnos` https://cclib.github.io/data_notes.html#atomcoords

Reply:
You can use [this](https://github.com/cclib/cclib/blob/1aedede314a594442c4248453f864725c225c298/cclib/parser/utils.py#L221) to get the atomnos from the symbols